pkg/tsdb/elasticsearch/client/client.go

```diff
@@ -150,7 +153,7 @@ func (c *baseClientImpl) encodeBatchRequests(requests []*multiRequest) ([]byte,
 	return payload.Bytes(), nil
 }
 
-func (c *baseClientImpl) executeRequest(method, uriPath string, body []byte) (*http.Response, error) {
+func (c *baseClientImpl) executeRequest(method, uriPath string, body []byte) (*SearchRequestInfo, *http.Response, error) {
```

Comment:
It looks a bit messy with 3 response values, is there a way to create a custom http.Response type (with http.Response embedded & SearchRequestInfo embedded) ?

Reply:
Thanks for the review. Yes there's always a way :smile: Thought it didn't matter that much since this was an internal method, but can look into it.

Reply:
Can put the `*SearchRequestInfo` and `*http.Response` in a struct if that's better?

Reply:
I don't think it matter a ton either, since as you say it's an internal method. But worth spending maybe 10min and checking if a struct with embedding is easy to switch to.

Reply:
Pushed a commit now using a struct embedding req info and http response. Think that is a better solution :+1: Whay do you think?